### ISSUE TITLE
Inconsistent results from {{ uri.rootUrl }}

### ISSUE THREAD

**Issue by withindale** [getgrav/grav#2667]
Hope this is the right place for the question.

Can the the following be corrected in settings or is it a GRAV issue?

On a page by itself {{ uri.rootUrl(true)/about}} gives the correct result: 
'https://www.cw.biz/grav/about' 

On the same page in /grav/site <a href="{{uri.rootUrl(true)}}/about> gives an extended result 👍 'https://www.cw.biz/grav/site/%7B%7Buri.rootUrl(true)%7D%7Dabout'.
This is the correct result from {{ uri.rootUrl }} concatenated with 'site/%7B%7Buri.rootUrl(true)%7D%7D'

When included in a CustomHTML particle the same statement <a href="{{uri.rootUrl(true)}}/about>
gives the correct result: 'https://www.cw.biz/grav/about' as above.

NOTES
1. The aim is to place a button at the bottom of web pages with a Custom HTML particle after the Content particle.
2. The original plan was to put the button in a shortcode. This worked on the page but rge Custom HTML did not process when shortcode processing was checked.
3. On the page {{ site.buttonText }} retrieved 'Button Text' but the Custom HTML particle did not.

**Comment by withindale** [comment]
In summary {{ rootUri() }} returns the correct string in a Twig file or a CustomHTML block. In Page content it appends a string. This appears to be the path to the variable containing the result, e.g. correctString/site/forms/contact/$s, where /site/forms/contact/$s is the appended string.

**Comment by mahagr** [comment]
Gantry has a bit different logic because it was built independently from Grav. So particles behave in a different way to Grav.

Though in this case, it looks like that twig isn't being parsed in the second case.

**Comment by withindale** [comment]
Yes, recognising the difference between Gantry and GRAV helps. Main issue
is the path in the return string.
Thanks for the note.

Ian Russell
